### PR TITLE
fix: remove rejected cache promise from pending list. 

### PR DESCRIPTION
**Pull request by tdekoning**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/18926

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolved https://github.com/nuxt/nuxt/issues/18926.

Cached events that have rejected were kept inside the `pending` list, causing later requests to the same event to reject as well. This led to events on the same path to never recover, because the code of the event was never called again after an error.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

File: playground/routes/error-route.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default defineCachedEventHandler(
```
Comment (manniL) [inline playground/routes/error-route.ts:1]: We sould probably remove the test route from the playground file 😋

Comment (tdekoning) [reply playground/routes/error-route.ts]: Done! Thanks for the suggestion

**Review by manniL — APPROVED** [review]

**Comment by tdekoning** [comment]
@pi0 do you have time to review this change? It seems to me as quite a small fix for a very impactful problem.

File: src/runtime/cache.ts [hunk 2]
```diff
@@ -90,18 +90,26 @@ export function defineCachedFunction<T = any>(
...
+          delete pending[key];
+          if (validate(entry)) {
+            useStorage()
+              .setItem(cacheKey, entry)
+              .catch((error) => console.error("[nitro] [cache]", error));
+          }
         }
+      } catch (error) {
+        // Make sure entries that reject get removed.
+        delete pending[key];
```
Comment (pi0) [inline src/runtime/cache.ts:109]: I think we could limit scope of try catch around L94 only.

**Comment by pi0** [comment]
Thanks for the PR dear @tdekoning and sorry took long to review.

**Pull request merged**